### PR TITLE
feat: allow multiple lines in alert component

### PR DESCRIPTION
**Pull request by saiprabhu-dandanayak**
### Fixes : [#30851](https://github.com/appsmithorg/appsmith/issues/30851) 

I have raised this pr in-order to allow multiple lines in alert message

### passing `\n` in the message
Screenshot from 2024-09-05 13-51-43

### unable to parse `\n`
Screenshot from 2024-09-05 13-50-06

### After issue is resolved
Screenshot from 2024-09-05 13-55-01

### Cypress Testing video

https://github.com/user-attachments/assets/ec826cef-664f-4392-b81e-9974178e65ec

## Summary by CodeRabbit

- **New Features**
	- Introduced a new styled component `StyledPre` for enhanced formatting in toast notifications.
	- Enhanced the Toast component to better render content, improving readability by preserving whitespace and formatting.
	- Added automated tests for the button widget's alert functionality, ensuring correct display of messages with newline characters.

- **Bug Fixes**
	- Improved the display of alert messages in toast notifications, ensuring they respect original formatting.

### CODE REVIEW

**Comment by saiprabhu-dandanayak** [comment]
**@appsmithorg/query-js-pod** , Please review this pr if you have enough bandwidth.
Thanks in advance.

**Comment by rishabhrathod01** [comment]
@saiprabhu-dandanayak Could you please fix the lint and prettier checks that are failing for the PR?

**Comment by saiprabhu-dandanayak** [comment]
Hi @**rishabhrathod01** , fixed the `lint` and `prettier` checks ,  could you pls check ??

**Comment by saiprabhu-dandanayak** [comment]
Hi **@rishabhrathod01** , if you have enough bandwidth , can you pls review this pr , i would be happy to work on comments if any.
Thanks in advance.

**Comment by NilanshBansal** [comment]
@carinanfonseca can you please check the design changes here

**Comment by carinanfonseca** [comment]
hello @saiprabhu-dandanayak
thank you for your contribution. 

It makes sense to recognize the \n when we are in JS mode, but in non-JS mode, a user shouldn't be expected to use \n to create a new line. In that mode, pressing 'Enter' in the text area itself should work, like this:
Screenshot 2024-09-20 at 14 57 56

let me know if this is possible. I am happy to answer any questions.

**Comment by saiprabhu-dandanayak** [comment]
Hi **@carinanfonseca**  , the screenshot i have added in PR description have double `\n` , so the spacing is more and even in non-JS mode if we provide  `Enter` it will work as expected , In non-JS mode even if we provide `\n` it is being resolved automatically.
Please let me know this is expected behaviour or not.

**Comment by rishabhrathod01** [comment]
@saiprabhu-dandanayak could you fix the storybook checks that are failing for the PR?

**Comment by saiprabhu-dandanayak** [comment]
Hi **@rishabhrathod01** , The error is showing it lacks ` Project token` and i am not sure how to fix this.
Any suggestions would be helpfull.
Thank You.

**Comment by rishabhrathod01** [comment]
@saiprabhu-dandanayak you could ignore the storybook failure for now. The Appsmith team will forward the designs for the Toast UI to be implemented here. 

cc: @carinanfonseca

**Comment by rishabhrathod01** [comment]
@saiprabhu-dandanayak could merge the latest release branch to the PR? Will need trigger the workflow again.

**Review by rishabhrathod01 — COMMENTED** [review]
Code changes look good. Will need to create duplicate PRs for the cypress test run.

**Comment by saiprabhu-dandanayak** [comment]
Hi @rishabhrathod01 , took recent pull from `release` branch and updated the source branch , pls check.
Thank you.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_showAlert_multiline_message_spec.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  { tags: ["@tag.Widget", "@tag.Button"] },
+  () => {
+    before("Login to the app and navigate to the workspace", function () {
+      entityExplorer.DragDropWidgetNVerify(draggableWidgets.BUTTON);
+    });
+
+    it("Verify showAlert message renders correctly with newline characters", () => {
+      propPane.EnterJSContext("onClick", "showAlert(`sai\n\nprabhu`)");
+      agHelper.ClickButton("Submit");
+      cy.wait(500);
```
Comment (rishabhrathod01) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_showAlert_multiline_message_spec.ts:19]: Could you remove the wait as suggested below? Does the test fails without wait?

Comment (saiprabhu-dandanayak) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_showAlert_multiline_message_spec.ts]: resolved, the test doesn't fail after removing cy.wait() , verified by running it.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_showAlert_multiline_message_spec.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      cy.get(".Toastify", { timeout: 1000 }).should(
+        "have.text",
+        "sai\n\nprabhu",
+      );
+    });
```
[the remark this answers is not in the document]
Comment (saiprabhu-dandanayak) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_showAlert_multiline_message_spec.ts]: resolved

**Comment by rishabhrathod01** [comment]
Shadow PR for Cypress run 
- https://github.com/appsmithorg/appsmith/pull/36514
- https://github.com/appsmithorg/appsmith-ee/pull/5218

File: app/client/packages/design-system/ads/src/Toast/Toast.tsx [hunk 3]
```diff
@@ -40,7 +40,7 @@ const toast = {
 
     return toastifyToast(
       <ToastBody kind="body-m">
-        {content}
+        <pre style={{ font: "inherit" }}>{content}</pre>
```
Comment (rishabhrathod01) [inline app/client/packages/design-system/ads/src/Toast/Toast.tsx:43]: Please remove the inline styling here and instead using something like
```js
// Styled component approach
const StyledPre = styled.pre`
  font: inherit;
`;
```


```jsx
<StyledPre> {context}</StyledPre>
```

**Review by rishabhrathod01 — CHANGES_REQUESTED** [review]

File: app/client/packages/design-system/ads/src/Toast/Toast.tsx (hunk 3 above)
Comment (saiprabhu-dandanayak) [reply app/client/packages/design-system/ads/src/Toast/Toast.tsx]: resolved

**Comment by rishabhrathod01** [comment]
Test has passed for the shadow PR - https://github.com/appsmithorg/appsmith/pull/36514 

This PR is safe to merge

**Review by rishabhrathod01 — APPROVED** [review]
Changes look good to me. 

Both CE and EE shadow PR ci-test run has passed. 
CE - https://github.com/appsmithorg/appsmith/pull/36514
EE - https://github.com/appsmithorg/appsmith-ee/pull/5218

Contributor made linting and prettier changes after that and the lint check has passed for the same. 

This PR is ready to be merged.

**Comment by albinAppsmith** [comment]
@saiprabhu-dandanayak  We are reverting this PR since it is causing below issues.

1. There was a white space in front of the content which was fixed by https://github.com/appsmithorg/appsmith/pull/36629
2. Content is not getting wrapped. https://github.com/appsmithorg/appsmith/issues/36642
3. While running this change in storybook, the toast is expanded in height.
Screenshot 2024-10-02 at 10 49 09

**Pull request merged**